applications/GeoMechanicsApplication/custom_strategies/strategies/geo_mechanics_newton_raphson_erosion_process_strategy.hpp

```diff
@@ -327,6 +381,25 @@ class GeoMechanicsNewtonRaphsonErosionProcessStrategy
         return GeoMechanicsNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>::SolveSolutionStep();
     }
 
+
+
+    /**
+     * @brief Collects the current solution vector for the degrees of freedom (DOFs).
+     * @details This method retrieves the current solution values for the provided DOF set.
+     * The solution vector will be resized to match the size of the DOF set if necessary,
+     * and will be filled with the solution values corresponding to each DOF. Each value is accessed
+     * using the equation ID associated with each DOF.
+     */
+    void CollectCurrentSolution() {
+        auto& rDofSet = this->GetBuilderAndSolver()->GetDofSet();
+        Vector this_solution(rDofSet.size());
+        block_for_each(rDofSet, [&](const auto& r_dof) {
```

Comment:
Could you explicitly capture the variables needed in the lambda expression here, instead of capturing everything implicitly?
```suggestion
        block_for_each(rDofSet, [&this_solution](const auto& r_dof) {
```